### PR TITLE
build(dev): Add frontend-only build for local development

### PR DESCRIPTION
**Pull request by billyvg**
Adds support for frontend-only local development against a remote API. `yarn dev-ui` to start-up the local dev server that proxies to `sentry.io`.

### CODE REVIEW

**Comment by billyvg** [comment]
I would like to test this internally a bit before "releasing" this.

File: package.json [hunk 1]
```diff
@@ -189,7 +189,7 @@
     "lint": "yarn eslint tests/js src/sentry/static/sentry/app --ext .js,.jsx,.ts,.tsx",
     "lint:css": "yarn stylelint 'src/sentry/static/sentry/app/**/*.[jt]sx'",
     "dev": "(yarn check --verify-tree || yarn install --check-files) && sentry devserver",
-    "dev-server": "webpack-dev-server",
+    "dev-ui": "SENTRY_UI_DEV_ONLY=1 SENTRY_WEBPACK_PROXY_PORT=7999 yarn webpack-dev-server",
```
Comment (billyvg) [inline package.json:192]: Thinking we use a different port to not conflict with reverse proxy? Any suggestions?

File: webpack.config.js [hunk 2]
```diff
@@ -379,17 +379,6 @@ let appConfig = {
...
 
-if (SENTRY_EXPERIMENTAL_SPA) {
-  /**
-   * Generate a index.html file used for running the app in pure client mode.
-   * This is currently used for PR deploy previews, where only the frontend
-   * is deployed.
-   */
-  appConfig.plugins.push(
-    new CopyPlugin([{from: path.join(staticPrefix, 'app', 'index.html')}])
-  );
```
Comment (billyvg) [inline webpack.config.js:390]: Removed in favor of `html-webpack-plugin`

File: webpack.config.js [hunk 3]
```diff
@@ -434,34 +425,71 @@ if (FORCE_WEBPACK_DEV_SERVER || (HAS_WEBPACK_DEV_SERVER_CONFIG && !NO_DEV_SERVER
...
-    ? appConfig.devServer
-    : {
-        ...appConfig.devServer,
-        before: () => undefined,
-        publicPath: '/_assets',
-        proxy: {'/api/': backendAddress},
-        historyApiFallback: {
-          rewrites: [{from: /^\/.*$/, to: '/_assets/index.html'}],
-        },
-      };
```
Comment (billyvg) [inline webpack.config.js:464]: This previously wasn't really used except to test the deploy preview configuration locally, but has been adapted to proxy requests to `sentry.io`

File: src/sentry/static/sentry/index.ejs [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
+  <!--[if lt IE 7 ]> <html lang="en" class="ie6" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if IE 7 ]>    <html lang="en" class="ie7" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if IE 8 ]>    <html lang="en" class="ie8" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if IE 9 ]>    <html lang="en" class="ie9" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if (gt IE 9)|!(IE)]><!--><html lang="en" class="" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <!--<![endif]-->
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/index.ejs:6]: Do we.. actually need any of this?

Comment (billyvg) [reply src/sentry/static/sentry/index.ejs:6]:
```suggestion
  <html lang="en">
```

File: package.json (hunk 1 above)
Comment (EvanPurkhiser) [reply package.json:192]: Yeah I think this is a good idea. Annoying to just randomly pick ports but 👍

**Review by evanpurkhiser — APPROVED** [review]
Can't wait to try this out.

File: src/sentry/static/sentry/index.ejs [hunk 5]
```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
+  <!--[if lt IE 7 ]> <html lang="en" class="ie6" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if IE 7 ]>    <html lang="en" class="ie7" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if IE 8 ]>    <html lang="en" class="ie8" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <!--[if IE 9 ]>    <html lang="en" class="ie9" <% if(htmlWebpackPlugin.files.manifest) { %> manifest="<%= htmlWebpackPlugin.files.manifest %>"<% } %>> <![endif]-->
+  <html lang="en">
```
Comment (billyvg) [inline src/sentry/static/sentry/index.ejs:6]:
```suggestion
  <html lang="en">
```

**Review by markstory — COMMENTED** [review]
Looking forward to this :+1:

**Pull request merged**